web_search_with_and/static/src/js/search_bar.esm.js

```diff
@@ -0,0 +1,23 @@
+/** @odoo-module **/
+
+import {patch} from "@web/core/utils/patch";
+import {SearchBar} from "@web/search/search_bar/search_bar";
+
+patch(SearchBar.prototype, {
+    selectItem(item) {
```

Comment:
This method seems to have changed in Odoo 17: https://github.com/odoo/odoo/blob/f1f8cbf/addons/web/static/src/search/search_bar/search_bar.js#L341-L348
Given that the method is overwritten without calling super, you'd have to incorporate the new code in this module.

Reply:
@jcadhoc I think you need to change this, because in Odoo 16 the method looked like this: 

```
    selectItem(item) {
        if (!item.unselectable) {
            const { searchItemId, label, operator, value } = item;
            this.env.searchModel.addAutoCompletionValues(searchItemId, { label, operator, value });
        }
        this.resetState();
    }
```
And in this module this parameter was added:  isShiftKey: this.isShiftKey to the addAutoCompletionValues call.

And because in Odoo 17 the method is like this:

```
    selectItem(item) {
        const searchItem = this.getSearchItem(item.searchItemId);
        if (
            (searchItem.type === "field" && searchItem.fieldType === "properties") ||
            (searchItem.type === "field_property" && item.unselectable)
        ) {
            this.toggleItem(item, !item.isExpanded);
            return;
        }

        if (!item.unselectable) {
            const { searchItemId, label, operator, value } = item;
            this.env.searchModel.addAutoCompletionValues(searchItemId, { label, operator, value });
        }
        this.resetState();
    }
```
You need to replace the method with the new code and add the isShiftKey: this.isShiftKey to the addAutoCompletionValues call.

In general overwriting methods without calling super is problematic, but in this case I do not have a better solution. But I am also not a Odoo JS expert.